inspire/base/static/js/search/search_results.js

```diff
@@ -44,8 +44,188 @@ require(['jquery', 'bootstrap'], function($) {
+          'Select all ' + $('#total-results').text() + ' results.</a></div></div>');
+      }
+
+      $('#select-all-records').on('click', function() {
+        $('#info-message').remove();
+        $.get("/search?of=id&rg=" + $('#total-results').text(), function(data, status) {
+          if (status == 'success') {
+            if (data.length > 5000) {
+              data.splice(5000, data.length);
+              $('#results-panel').after('<div class="alert alert-warning" id="alert-selection" role="alert">' +
+                '5000 records have been selected.<a class="pointer" id="undo-selection"> Undo selection.</a></div>');
+            } else {
+              $('#results-panel').after('<div class="alert alert-warning" id="alert-selection" role="alert">' +
+                $('#total-results').text() + ' records have been selected.<a class="pointer" id="undo-selection"> Undo selection.</a></div>');
+            }
+            sList = data;
+          }
+        });
+      });
+    } else {
+      $('.checkbox-results').each(function() {
+        this.checked = false;
+      });
+      sList = [];
+      $('#info-message').remove();
+      $('#alert-selection').remove();
+    }
+  });
+
+  $(document).on('click', '#undo-selection', function(e) {
+    sList = [];
+    $('#alert-selection').remove();
+    $('.checkbox-results').each(function() {
+      this.checked = false;
+    });
+    $('#checkbox-select-all').prop('checked', false);
+  });
+
+
+  $('#checkbox-parent > input[type=checkbox]').change(function() {
+    var $this = $(this);
+    if (this.checked) {
+      sList.push($this.attr("id"));
+    } else {
+      var index = sList.indexOf($this.attr("id"))
+      if (index > -1) {
+        sList.splice(index, 1);
+      }
+    }
+    if (sList != 0) {
+      $("#dropdown-export").html($('#dropdown-export').text().trim() + ' <span class="caret"></span>');
+    }
+  });
+
+  var $loading = $('#spinner-download').hide();
+  $("#download-format").on('click', function() {
+    url = create_url($('#dropdown-export').text().trim());
+    var uniqueIds = filter_list(sList);
+    if (uniqueIds.length != 0) {
+      $loading.show();
+      $('#download-format').text('Downloading...');
+      $('#download-format').addClass('disabled');
+      $.ajax({
+        type: "POST",
+        url: url,
+        data: {
+          ids: uniqueIds
+        },
+        success: function(data) {
+          var response_data = "text/plain;charset=utf-8," + encodeURIComponent(data);
+          $("body").append('<a id="data-download" href="data:' + response_data + '" download="export_as.' +
+            get_format($('#dropdown-export').text().trim()) + '">download</a>');
+          var trigger_element = document.getElementById('data-download');
+          trigger_element.click();
+          $("#data-download").remove();
+        }
+      });
+    }
+  });
+
+
+  $(document)
```

Comment:
Instead of this, use `.done()` in the AJAX request (see http://api.jquery.com/jquery.ajax/)